### ISSUE TITLE
GETJOY does not populate non-directional buttons

### ISSUE THREAD

**Issue by SlithyMatt** [commanderx16/x16-emulator#200]
With the change to r34, the memory addresses populated by GETJOY changed, and the directional controls are getting set correctly in the new location, whether using an actual controller or the keyboard cursor keys. However, the buttons (start, select, A, B) don't have any effect on the new location, whether using keyboard (alt, ctrl, space, enter) or an actual controller.

**Comment by SlithyMatt** [comment]
Not reproduced using new test program on another machine. Will test new program on machine where issue first found with both keyboard and USB controller. Will recommend issue deletion if not reproduced. Otherwise, will have more detailed information on particular failure case.

**Comment by SlithyMatt** [comment]
Not reproducible on original system where failure happened using new program. Old program did not pause for interrupts before calling GETJOY. This was not an issue with R33 and older. With R34, adding a WAI followed by a check for a VSYNC detected in the last IRQ fixes the problem. For some reason, the "buttons" are more sensitive to this than the directionals. I noticed brief flashing while holding down buttons, indicating that the main loop running too fast was causing the button press to be cleared. Meanwhile, the directionals stayed pressed in the register even without waiting for interrupts. Not sure if this is a real issue at this point, because basic good programming practice is a "workaround". Closing issue because of this.
